yield_plugin/src/yield_plugin.cpp

```diff
@@ -576,12 +581,24 @@ namespace yield_plugin
 
         // Calculate the distance between the two interpolated points
         double distance = std::sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
+
         smallest_dist = std::min(distance, smallest_dist);
         RCLCPP_DEBUG_STREAM(nh_->get_logger(), "Smallest_dist: " << smallest_dist << ", distance: " << distance << ", dt: " << dt
           << ", x1: " << x1 << ", y1: " << y1
           << ", x2: " << x2 << ", y2: " << y2
           << ", p2a_t:" << std::to_string(p2a_t));
 
+        // Following "if logic" assumes the traj2 is a simple cv model, aka, traj2 point is a straight line over time.
+        // And current traj1 point is fixed in this iteration.
+        // Then once the distance between the two start to increase over traj2 iteration,
+        // the distance will always increase and it's unnecessary to continue the logic to find the smallest_dist
+        if (previous_distance_between_predictions < distance)
+        {
+          RCLCPP_DEBUG_STREAM(nh_->get_logger(), "Stopping search here because the distance between predictions started to increase");
+          continue;
```

Comment:
This is going to continue the next nested `for` loop's iteration. I believe we want `break` so we move on to the outer loop's next iteration.

Reply:
Applied this change and overall it got better but didn't fix the worst case scenario. Sometimes nested for loop still requires as much as `0.211` sec ti complete, which I addressed by dynamically skipping number of predicted states in the list worth of 1.5 meters.